src/jokes/utils/urls.py

```diff
@@ -0,0 +1,29 @@
+from typing import Mapping
+from enum import Enum
+from jokes.utils.params import GetEndpointParams, SubmitEndpointParams
+
+
+BASE_URL = "https://v2.jokeapi.dev"
+
+Primitive = str | int | float | bool | None
+TParam = GetEndpointParams | SubmitEndpointParams
+
+
+class Endpoints(Enum):
+    GET = "joke"
+    SUBMIT = "submit"
+
+
+def format_params(dict: Mapping[str, Primitive]) -> str:
```

Comment:
Passing params here and then calling the `dict()` method internally is possible, but it doesn't ensure the safety of the join, specifically the `f"{k}={v}"` since the pydantic dict method only requires a `dict[str, Any]`.

So theoretically we could construct a parameter class that returns a dict of type `dict[str, SomeComplexType]` which would either break the query string or return something illegible.